build.sh

```diff
@@ -12,12 +12,22 @@ shift "$(( OPTIND - 1 ))"
 
 NAME=$(echo "${PWD##*/}" | tr _ -)
 TAG="latest"
+MULTISTAGE_TARGET="development"
+
+UID="$(id -u "${USER}")"
```

Comment:
Sorry to be _that mac guy_, but I think it's worth making this cross-platform at this stage.

The problem is two-fold:
1. On mac, the `UID` and `GID` environment variables are read-only, as they are already produced by the system. Therefore the local variable would need to be renamed (for example MYUID)
2. If the mac user is part of the "staff" user group (root), the GID will be 20. In ubuntu this is a reserved group ("dialout"), so the docker build step `addgroup --gid ${GID}` will fail.

Conversely though, mac also resolves the whole permissions thing nicely in a different way:

> Inside the container, the osxfs driver pretends that whatever uid/gid is running in the container is also the uid/gid that owns the mounted files.
> If you chown files (in the container) to something else, no chown is performed on the real files; this owner information is stored in an extended file attribute instead, and that value is used by the container (and ignored by macOS).
> The real files are owned by whoever owns them in macOS, outside the container. Access control is determined using those real file ownerships, and the uid/gid of the user running the Docker applications (which is probably your login user on the Mac)

So in other words, this UID / GID mapping between host and container is actually only necessary on linux in the first place.


In conclusion, if you do the following, this build script will work on both linux and mac without issue.

```
MYUID="$(id -u "${USER}")"
MYGID="$(id -g "${USER}")"

if [ "$(uname -s)" = "Darwin" -a "${MYGID}" = 20 ]; then
    MYGID=1000
fi
```

Reply:
perfect. Can you add that? I don't have a mac so I can't test that but I totally trust you :D

Reply:
Sure, I tested it already and it worked so I will push a small commit with just this change. I'll leave you to resolve the remaining comments after